### DISCUSSION TITLE
Drawer not rendering correctly in v3

### DISCUSSION THREAD

**Discussion by EliasLeguizamon123** [chakra-ui/chakra-ui#9173] [Q&A] [answered]
I just migrated to `Chakra V3`, and I am encountering several issues with the `Drawer` component.

I am currently working on a project using Vite, React, and TypeScript.

I can't install the Chakra snippet because I get multiple typing errors in the component downloaded via Chakra CLI.

When I try to use the component imported from `@chakra-ui/react`, it looks very strange. It doesn't take up the full height it should; instead, it only occupies the full height of the parent container (this didn't happen before). I'm trying to use it with the following code, which worked correctly in Chakra version 2 but no longer works in the new version 3.

```jsx

// Father component
export default function Navbar() {
    const screenHeight = useScreenHeight();

    return (
        <Flex
            align={'center'}
            bgColor="#0051AD"
            color="white"
            direction={'row'}
            fontSize={'1.3rem'}
            height={screenHeight * 0.23}
            justify="center"
            justifyContent={[
                'space-between',
                'space-between',
                'center',
                'center',
            ]}
            minH={['150px', '160px']}
            px={'1rem'}
            py={'0.5rem'}
            top={'0'}
            w="100vw"
            zIndex="1000"
        >
            <DrawerMenu />
        </Flex>
    );
}

```

```jsx
// new drawer of chakra v3 (tacked from docs of chakra v3)
import {
    Button,
    DrawerRoot,
    DrawerActionTrigger,
    DrawerBackdrop,
    DrawerBody,
    DrawerCloseTrigger,
    DrawerContent,
    DrawerFooter,
    DrawerHeader,
    DrawerTitle,
    DrawerTrigger,
} from '@chakra-ui/react';
import { useState } from 'react';

export default function DrawerMenu() {
    const [open, setOpen] = useState(false);

    return (
        <DrawerRoot open={open} onOpenChange={(e: any) => setOpen(e.open)}>
            <DrawerBackdrop />
            <DrawerTrigger>
                <Button size="sm" variant="outline">
                    Open Drawer
                </Button>
            </DrawerTrigger>
            <DrawerContent>
                <DrawerHeader>
                    <DrawerTitle>Drawer Title</DrawerTitle>
                </DrawerHeader>
                <DrawerBody>
                    <p>
                        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                        Sed do eiusmod tempor incididunt ut labore et dolore
                        magna aliqua.
                    </p>
                </DrawerBody>
                <DrawerFooter>
                    <DrawerActionTrigger asChild>
                        <Button variant="outline">Cancel</Button>
                    </DrawerActionTrigger>
                    <Button>Save</Button>
                </DrawerFooter>
                <DrawerCloseTrigger />
            </DrawerContent>
        </DrawerRoot>
    );
}

```

This is the original view

![image](https://github.com/user-attachments/assets/19711c36-451a-4876-81c4-052fa4ae940f)

And this is the open drawer view

![image](https://github.com/user-attachments/assets/e05586f2-8dba-4194-bcb1-e9edc1f5a08f)

- Am I doing something wrong?

- What changes should I make to fix how the Drawer currently looks?

- Has anyone experienced something similar to what I’m facing?

**Comment by segunadebayo** [comment, marked as answer]
You can try installing just the drawer snippet if that helps.

```sh
npx chakra snippets add drawer
```
---

The fix for the code you shared above is to wrap the `DrawerContent` within `DrawerPositioner` (import it from chakra)

**Reply by EliasLeguizamon123** [reply]
Hi @segunadebayo! thanks for your answer, 

> The fix for the code you shared above is to wrap the DrawerContent within DrawerPositioner (import it from chakra)

With this, I was able to fix the visual issue. Thank you so much! 🚀 


> You can try installing just the drawer snippet if that helps.

With the new snippets, I am encountering several problems.

For example, the drawer snippet has many typing issues. If I take a look at the component, this is what happens:

![image](https://github.com/user-attachments/assets/186879ec-ba0f-4520-9268-9a4a376c55e8)

These errors describe the following: 

```typescript
Property 'padding' does not exist on type 'DrawerContentProps'.

Property 'children' does not exist on type 'DrawerContentProps'.

Type '{ children: Element; padding: any; }' is not assignable to type 'IntrinsicAttributes & DrawerPositionerProps & RefAttributes<HTMLDivElement>'.
  Property 'children' does not exist on type 'IntrinsicAttributes & DrawerPositionerProps & RefAttributes<HTMLDivElement>'.
  
Type '{ children: any; asChild: boolean; ref: ForwardedRef<HTMLDivElement>; }' is not assignable to type 'IntrinsicAttributes & DrawerContentProps & RefAttributes<HTMLDivElement>'.
  Property 'children' does not exist on type 'IntrinsicAttributes & DrawerContentProps & RefAttributes<HTMLDivElement>'.
  
Type '{ children: Element; asChild: true; insetEnd: string; position: string; top: string; }' is not assignable to type 'IntrinsicAttributes & DrawerCloseTriggerProps & RefAttributes<HTMLButtonElement>'.
  Property 'children' does not exist on type 'IntrinsicAttributes & DrawerCloseTriggerProps & RefAttributes<HTMLButtonElement>'.
```

Additionally, I migrated from Chakra v2 to v3 following the migration guide provided in the documentation: 
- [Using Chakra in Vite](https://www.chakra-ui.com/docs/get-started/frameworks/vite).

- [Migration to v3](https://www.chakra-ui.com/docs/get-started/migration).

Despite following the steps, I am still encountering these issues. Could you help me identify if there's something missing or if this is a known issue?
